python/django/datasources/views.py

```diff
@@ -174,6 +176,21 @@ class DemographicDataFeatureViewSet(OTIAdminViewSet):
     filter_fields = ('datasource',)
 
 
+class DemographicDataRanges(APIView):
+    """ Endpoint to GET the min/max ranges of demographic data
+
+    GET params:
+      type: String (required), Name of the demographic data field to use for range calculation
+    """
+    def get(self, request, *args, **kwargs):
+        get_type = request.QUERY_PARAMS.get('type', None)
+        if not get_type:
+            return Response("type parameter is required", status=status.HTTP_200_OK)
```

Comment:
This should be a 4xx code; I think changing this may let you simplify getDemographicRange as well.